File: leetcode/strings/dart/maximum_odd_binary_number.dart
class Solution {
  String maximumOddBinaryNumber(String s) {
    var counter = 0;
    for (var i = 0; i < s.length; i++) {
      if (s[i] == "1") {
        counter++;
      }
    }
    return "1" * (counter - 1) + "0" * (s.length - counter) + "1";
  }
}

void main(List<String> args) {
  final obj = Solution();
  final s = '101001';
  print(obj.maximumOddBinaryNumber(s));
}

File: leetcode/strings/go/maximum_odd_binary_number.go
package main

import "strings"

func maximumOddBinaryNumber(s string) string {
	counter := 0
	for i := 0; i < len(s); i++ {
		if s[i] == '1' {
			counter++
		}
	}
	return strings.Repeat("1", counter-1) + strings.Repeat("0", len(s)-counter) + "1"

}
